File: nodeserver/tracker.js
var PUBLISH_TIMEOUT = 10;
var ALT_URLS = {};
var FRIENDS = [];

function publish(gameId, portNumber)
{
}

function unpublish(port)
{
}

function publishTimeout()
{
    rpc.response(PUBLISH_TIMEOUT);
}

function listFirst(gameId)
{
}

function list(gameId, offset)
{
}

function verify(address, port)
{
}

function altUrls()
{
    rpc.response(ALT_URLS);
}

function friends()
{
    rpc.response(FRIENDS);
}

exports.publish = publish;
exports.unpublish = unpublish;
exports.publishTimeout = publishTimeout;
exports.listFirst = listFirst;
exports.list = list;
exports.verify = verify;
exports.altUrls = altUrls;
exports.friends = friends;

File: server/tracker.cpp
#include "tracker.h"

Tracker &Tracker::instance()
{
    static Tracker *tracker = new Tracker;
    return *tracker;
}

void Tracker::publish(const QString &gameId, const QHostAddress &address,
                      quint16 port, const QVariantMap &options)
{
    // TODO:
    //  * Add game room to the list
    //   * Only one game room can exist by address/ip
    //  * Remove the room from the list when it reachs timeout
}

void Tracker::unpublish(const QHostAddress &address, quint16 port)
{
    // TODO:
    // * Remove game room from list
}

int Tracker::publishTimeout()
{
    return 2;
}

QVariantMap Tracker::listFirst(const QString &gameId,
                               const QVariantMap &options)
{
    // TODO
}

QVariantMap Tracker::list(const QString &gameId, const QVariantMap &options,
                          const QVariant &offset)
{
    // TODO
}

void Tracker::verify(const QHostAddress &address, quint16 port)
{
    // TODO
}

Tracker::Tracker() :
    QObject(NULL)
{
}
